lib/rdoc/code_object/any_method.rb

```diff
@@ -360,15 +360,25 @@ def deduplicate_call_seq(call_seq)
       ignore << is_alias_for.name
       ignore.concat is_alias_for.aliases.map(&:name)
     end
-    ignore.map! { |n| n =~ /\A\[/ ? /\[.*\]/ : n}
+
+    bracket_methods = ignore.grep(/\A\[/)
+    ignore -= bracket_methods
     ignore.delete(method_name)
-    ignore = Regexp.union(ignore)
 
-    matching = entries.reject do |entry|
-      entry =~ /^\w*\.?#{ignore}[$\(\s]/ or
-        entry =~ /\s#{ignore}\s/
+    if ignore.any?
+      ignore_union = Regexp.union(ignore)
+      entries.reject! do |entry|
+        entry =~ /^(\w*\.)?#{ignore_union}([\(\s]|$)/ or
+          entry =~ /\s#{ignore_union}\s/
```

Comment:
```suggestion
        /^(\w*\.)?#{ignore_union}([\(\s]|$)/.match?(entry) or
          \s#{ignore_union}\s/.match?(entry)
```

BTW:

* Can we use `\A`/`\z` not `^`/`$` here?
* Can we remove needless `\` escape from `[\(]`?
* How about using `(?:...)` not `(...)` not referred groups?